Sources/NIO/SocketOptionProvider.swift

```diff
@@ -12,272 +12,6 @@
-        return self.unsafeGetSocketOption(level: .ipv6, name: .ipv6_multicast_if)
-    }
-
-    /// Sets the socket option IPV6_MULTICAST_HOPS to `value`.
-    ///
-    /// - parameters:
-    ///     - value: The value to set IPV6_MULTICAST_HOPS to.
-    /// - returns: An `EventLoopFuture` that fires when the option has been set,
-    ///     or if an error has occurred.
-    public func setIPv6MulticastHops(_ value: CInt) -> EventLoopFuture<Void> {
-        return self.unsafeSetSocketOption(level: .ipv6, name: .ipv6_multicast_hops, value: value)
-    }
-
-    /// Gets the value of the socket option IPV6_MULTICAST_HOPS.
-    ///
-    /// - returns: An `EventLoopFuture` containing the value of the socket option, or
-    ///     any error that occurred while retrieving the socket option.
-    public func getIPv6MulticastHops() -> EventLoopFuture<CInt> {
-        return self.unsafeGetSocketOption(level: .ipv6, name: .ipv6_multicast_hops)
-    }
-
-    /// Sets the socket option IPV6_MULTICAST_LOOP to `value`.
-    ///
-    /// - parameters:
-    ///     - value: The value to set IPV6_MULTICAST_LOOP to.
-    /// - returns: An `EventLoopFuture` that fires when the option has been set,
-    ///     or if an error has occurred.
-    public func setIPv6MulticastLoop(_ value: CUnsignedInt) -> EventLoopFuture<Void> {
-        return self.unsafeSetSocketOption(level: .ipv6, name: .ipv6_multicast_loop, value: value)
-    }
-
-    /// Gets the value of the socket option IPV6_MULTICAST_LOOP.
-    ///
-    /// - returns: An `EventLoopFuture` containing the value of the socket option, or
-    ///     any error that occurred while retrieving the socket option.
-    public func getIPv6MulticastLoop() -> EventLoopFuture<CUnsignedInt> {
-        return self.unsafeGetSocketOption(level: .ipv6, name: .ipv6_multicast_loop)
-    }
-
-    #if os(Linux) || os(FreeBSD) || os(Android)
-        /// Gets the value of the socket option TCP_INFO.
-        ///
-        /// This socket option cannot be set.
-        ///
-        /// - returns: An `EventLoopFuture` containing the value of the socket option, or
-        ///     any error that occurred while retrieving the socket option.
-        public func getTCPInfo() -> EventLoopFuture<tcp_info> {
-            return self.unsafeGetSocketOption(level: .tcp, name: .tcp_info)
-        }
-    #endif
-
-    #if os(iOS) || os(macOS) || os(tvOS) || os(watchOS)
-        /// Gets the value of the socket option TCP_CONNECTION_INFO.
-        ///
-        /// This socket option cannot be set.
-        ///
-        /// - returns: An `EventLoopFuture` containing the value of the socket option, or
-        ///     any error that occurred while retrieving the socket option.
-        public func getTCPConnectionInfo() -> EventLoopFuture<tcp_connection_info> {
-            return self.unsafeGetSocketOption(level: .tcp, name: .tcp_connection_info)
-        }
-    #endif
-}
-
-
 extension BaseSocketChannel: SocketOptionProvider {
```

Comment:
Should we rename this file `BaseSocketChannel+SocketOptionProvider.swift` or  just move this extension to `BaseSocketChannel.swift`?